From 457d73d7b889f0c15a80f11681a4e79bf21feb3b Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Mon, 23 Sep 2019 14:34:40 +0800
Subject: [PATCH] drm/rockchip: driver: fix compile error when disable vop

drivers/gpu/drm/panel/panel-simple.c:385: undefined reference to
`rockchip_drm_crtc_send_mcu_cmd'

Change-Id: I95f182dca7abcb1e8d8a3fc3bb6fd20db5f2db65
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.h b/drivers/gpu/drm/rockchip/rockchip_drm_drv.h
index f68be755331f..7bd71f37343e 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.h
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.h
@@ -185,9 +185,18 @@ int rockchip_register_crtc_funcs(struct drm_crtc *crtc,
 				 const struct rockchip_crtc_funcs *crtc_funcs);
 void rockchip_unregister_crtc_funcs(struct drm_crtc *crtc);
 int rockchip_drm_wait_vact_end(struct drm_crtc *crtc, unsigned int mstimeout);
+#if IS_ENABLED(CONFIG_DRM_ROCKCHIP)
 int rockchip_drm_crtc_send_mcu_cmd(struct drm_device *drm_dev,
 				   struct device_node *np_crtc,
 				   u32 type, u32 value);
+#else
+static inline int rockchip_drm_crtc_send_mcu_cmd(struct drm_device *drm_dev,
+						 struct device_node *np_crtc,
+						 u32 type, u32 value)
+{
+	return 0;
+}
+#endif
 
 extern struct platform_driver cdn_dp_driver;
 extern struct platform_driver dw_hdmi_rockchip_pltfm_driver;
-- 
2.35.3

